experimental/cloudstrapper/playbooks/roles/cfn/cfnMagmaEksFargate.json

```diff
@@ -0,0 +1,294 @@
+         "CidrBlock" : { "Ref": "paramBridgeCidr" },
+         "AvailabilityZone" : { "Ref": "paramAvlZone1" },
+         "MapPublicIpOnLaunch": "true",
+         "Tags" : [ { "Key" : "Name", "Value" : { "Ref": "paramBridgeSubnetTagName" }} ]
+       }
+    },
+    "subPrivNet1" : {
+      "Type" : "AWS::EC2::Subnet",
+      "Properties" : {
+        "VpcId" : { "Ref" : "vpcMagmaEks" },
+        "CidrBlock" : { "Ref": "paramPriv1Cidr" },
+        "AvailabilityZone" : { "Ref": "paramAvlZone2" },
+        "Tags" : [ { "Key" : "Name", "Value" : { "Ref": "paramPriv1SubnetTagName" }} ]
+       }
+    },
+    "subPrivNet2" : {
+      "Type" : "AWS::EC2::Subnet",
+      "Properties" : {
+        "VpcId" : {  "Ref": "vpcMagmaEks" },
+        "CidrBlock" : { "Ref": "paramPriv2Cidr" },
+        "AvailabilityZone" : { "Ref": "paramAvlZone3" },
+        "Tags" : [ { "Key" : "Name", "Value" : { "Ref": "paramPriv2SubnetTagName" }} ]
+      }
+    },
+    "gwInetBridge" : {
+      "Type" : "AWS::EC2::InternetGateway",
+        "Properties" : {
+          "Tags" : [ {"Key" : "Name", "Value" : { "Ref" : "paramGwTagName"}} ]
+        }
+     },
+     "gwInetAttach" : {
+       "Type" : "AWS::EC2::VPCGatewayAttachment",
+         "Properties" : {
+           "VpcId" : { "Ref" : "vpcMagmaEks" },
+           "InternetGatewayId" : { "Ref" : "gwInetBridge" }
+         }
+    },
+    "eipNatGw" : {
+      "Type" : "AWS::EC2::EIP",
+      "DependsOn": "gwInetAttach",
+      "Properties" : {
+         "Domain" : "vpc"
+      }
+    },
+    "gwNatGw" : {
+      "Type" : "AWS::EC2::NatGateway",
+        "Properties" : {
+          "AllocationId" : { "Fn::GetAtt" : ["eipNatGw", "AllocationId"]},
+          "SubnetId" : { "Ref" : "subBridgeNet"},
+          "Tags" : [ {"Key" : "Name", "Value" : { "Ref" : "paramGwTagName"}} ]
+        },
+        "DependsOn": "subBridgeNet"
+    },
+    "rtBridgeInetGw" : {
+      "Type" : "AWS::EC2::RouteTable",
+      "Properties" : {
+        "VpcId" : { "Ref" : "vpcMagmaEks" },
+        "Tags" : [ { "Key" : "Name", "Value" : "Route Table for Bridge Network for Internet Gateway" } ]
+       }
+    },
+    "rteBridgeInetGw" : {
+      "Type" : "AWS::EC2::Route",
+      "Properties" : {
+        "RouteTableId" : { "Ref" : "rtBridgeInetGw" },
+        "DestinationCidrBlock" : "0.0.0.0/0",
+        "GatewayId" : { "Ref" : "gwInetBridge" }
+       }
+    },
+    "rtBridgeInetGwAssocSubnet" : {
+      "Type" : "AWS::EC2::SubnetRouteTableAssociation",
+      "Properties" : {
+        "SubnetId" : { "Ref" : "subBridgeNet" },
+        "RouteTableId" : { "Ref" : "rtBridgeInetGw" }
+       }
+    },
+    "sgVpc": {
+      "Type" : "AWS::EC2::SecurityGroup",
+      "Properties" : {
+        "GroupDescription" : "Allow all TCP to client host",
+	"GroupName": { "Ref": "paramSecGroupName" },
+        "VpcId" : {"Ref" : "vpcMagmaEks"},
+        "SecurityGroupIngress" : [{
+          "IpProtocol" : "-1",
+          "FromPort" : 0,
```

Comment:
I understand it is experimental for now.  I guess,  at some point we need to list the specific ports (originating from AGW) so that it is not wide open.  Perhaps we can document all ports used by AGW -> Orchestrator

Reply:
Agree. Individual implementations would have to choose which ports need to be locked and which ones need to open based on the services that are run and hopefully, this experimental feature provides a placeholder for all such rules. The container build script includes a list of all ports needed by the various services that run on the AGW and could provide the first template.